File: Lang - Zig/conditions.zig
const std = @import("std");
const print = std.debug.print;
const mem = std.mem;

pub fn main() !void {
    const PROMPT = "Please enter one of the following:";
    const TYPE = "type";
    const TS = "typescript";
    const JS = "javascript";
    const ANYTHING = "Or anything!!";

    print("{s}\n\t{s}\n\t{s}\n\t{s}\n\t{s}\n", .{ PROMPT, TYPE, TS, JS, ANYTHING });

    var buffer: [1024]u8 = undefined;
    const stdin = std.io.getStdIn().reader();

    // read the console, store the success or failure in result
    const result = try stdin.readUntilDelimiterOrEof(&buffer, '\n');

    if (result) |user_str| {
        if (user_str.len == 1) {
            print("That's nothing!\n", .{});
        } else if (user_str.len >= TS.len and (string_comparison(user_str, TS) or string_comparison(user_str, JS))) {
            print("Ew ew ew ew away!\n", .{});
        } else if (user_str.len >= TYPE.len and string_comparison(user_str, TYPE)) {
            print("safety!\n", .{});
        } else {
            print("Hmm yes I see, {s}\n", .{user_str[0..user_str.len]});
        }
    } else {
        print("Something went wrong\n", .{});
    }
}

// easier then mem.eql(u8, user_str[0..TYPE.len], TYPE)
// a little easier then that in every if statement
fn string_comparison(input: []u8, target: []const u8) bool {
    return mem.eql(u8, input[0..target.len], target);
}

File: Lang - C/conditions.c
#include <stdio.h>

#define PROMPT "Please enter one of the following"
#define TAB "\t"
#define TYPE "type"
#define TS "typescript"
#define JS "javascript"
#define ANYTHING "Or anything!"

int compare(char *str, char *smaller, size_t max_size) {
    for (int i = 0; i < max_size; i++) {
        if (str[i] == smaller[i]) {
            continue;
        }
        else {
            return 0;
        }
    }
    return 1;
}
int main(void) {
    printf("%s\n", PROMPT);
    printf("%s%s\n", TAB, TYPE);
    printf("%s%s\n", TAB, TS);
    printf("%s%s\n", TAB, JS);
    printf("%s%s\n", TAB, ANYTHING);

    char selection[64];
    scanf("%s", selection);

    if (compare(selection, ".", 2)) {
        printf("That's nothing!");
    }
    else if (compare(selection, TYPE, 5))
    {
        printf("Safety");
    }
    else if (compare(selection, TS, 11) || compare(selection, JS, 11))
    {
        printf("Ew ew ew ew away!");
    }
    else {
        printf("Hmm yes I see, %s", selection);
    }

    return 0;
}